xbmc/commons/Buffer.h

```diff
@@ -123,6 +123,15 @@ namespace XbmcCommons
       clear();
     }
 
+    /**
+    * Construct an buffer from std::vector<char>.
+    */
+    inline Buffer(const std::vector<char> &vec)
+      : Buffer(vec.size())
+    {
+      memcpy(buffer, vec.data(), vec.size());
+    }
+
```

Comment:
You don't need this method. Just use
```
  Buffer* buffer = new Buffer();
  buffer->put(vec.data(), vec.size());
```
This does a memcpy.

However, you could do this "zero copy" using:
```
Buffer buffer = new Buffer(vec.data(), vec.size());
```
The `std::vector<char>` is considered the owner of the memory. It needs to outlive the `Buffer.`